[Error] Internal: Uncaught exception thrown of type TokenMgrError (null)

hay,

I'm unable to compile https://github.com/runtimeverification/beacon-chain-spec/tree/v0.9.1. There might be some simple error, but this error message is completely inscrutable to me. Full stack trace:
```
org.kframework.parser.concrete2kore.disambiguation.TokenMgrError
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:598)
	at java.util.concurrent.ForkJoinTask.reportException(ForkJoinTask.java:677)
	at java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:735)
	at java.util.stream.ReduceOps$ReduceOp.evaluateParallel(ReduceOps.java:714)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
	at org.kframework.kompile.DefinitionParsing.resolveNonConfigBubbles(DefinitionParsing.java:280)
	at org.kframework.kompile.DefinitionParsing.lambda$resolveNonConfigBubbles$7(DefinitionParsing.java:250)
	at org.kframework.definition.ModuleTransformer.$anonfun$apply$1(transformers.scala:74)
	at scala.collection.concurrent.TrieMap.getOrElseUpdate(TrieMap.scala:903)
	at org.kframework.definition.ModuleTransformer.apply(transformers.scala:71)
	at org.kframework.definition.ModuleTransformer.apply(transformers.scala:67)
	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:234)
	at scala.collection.immutable.HashSet$HashSet1.foreach(HashSet.scala:320)
	at scala.collection.immutable.HashSet$HashTrieSet.foreach(HashSet.scala:976)
	at scala.collection.TraversableLike.map(TraversableLike.scala:234)
	at scala.collection.TraversableLike.map$(TraversableLike.scala:227)
	at scala.collection.AbstractSet.scala$collection$SetLike$$super$map(Set.scala:47)
	at scala.collection.SetLike.map(SetLike.scala:101)
	at scala.collection.SetLike.map$(SetLike.scala:101)
	at scala.collection.AbstractSet.map(Set.scala:47)
	at org.kframework.definition.ModuleTransformer.$anonfun$apply$1(transformers.scala:72)
	at scala.collection.concurrent.TrieMap.getOrElseUpdate(TrieMap.scala:903)
	at org.kframework.definition.ModuleTransformer.apply(transformers.scala:71)
	at org.kframework.definition.ModuleTransformer.apply(transformers.scala:67)
	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:234)
	at scala.collection.immutable.Set$Set3.foreach(Set.scala:167)
	at scala.collection.TraversableLike.map(TraversableLike.scala:234)
	at scala.collection.TraversableLike.map$(TraversableLike.scala:227)
	at scala.collection.AbstractSet.scala$collection$SetLike$$super$map(Set.scala:47)
	at scala.collection.SetLike.map(SetLike.scala:101)
	at scala.collection.SetLike.map$(SetLike.scala:101)
	at scala.collection.AbstractSet.map(Set.scala:47)
	at org.kframework.definition.ModuleTransformer.$anonfun$apply$1(transformers.scala:72)
	at scala.collection.concurrent.TrieMap.getOrElseUpdate(TrieMap.scala:903)
	at org.kframework.definition.ModuleTransformer.apply(transformers.scala:71)
	at org.kframework.definition.ModuleTransformer.apply(transformers.scala:67)
	at org.kframework.definition.DefinitionTransformer.apply(transformers.scala:110)
	at org.kframework.kompile.DefinitionParsing.resolveNonConfigBubbles(DefinitionParsing.java:250)
	at org.kframework.kompile.DefinitionParsing.parseDefinitionAndResolveBubbles(DefinitionParsing.java:176)
	at org.kframework.kompile.Kompile.parseDefinition(Kompile.java:147)
	at org.kframework.kompile.Kompile.run(Kompile.java:121)
	at org.kframework.kompile.KompileFrontEnd.run(KompileFrontEnd.java:70)
	at org.kframework.main.FrontEnd.main(FrontEnd.java:62)
	at org.kframework.main.Main.runApplication(Main.java:118)
	at org.kframework.main.Main.runApplication(Main.java:108)
	at org.kframework.main.Main.main(Main.java:56)
Caused by: org.kframework.parser.concrete2kore.disambiguation.TokenMgrError: Lexical error at line 1, column 2.  Encountered: "\"" (34), after : ""
	at org.kframework.parser.concrete2kore.disambiguation.SmtSortParserTokenManager.getNextToken(SmtSortParserTokenManager.java:311)
	at org.kframework.parser.concrete2kore.disambiguation.SmtSortParser.jj_ntk(SmtSortParser.java:194)
	at org.kframework.parser.concrete2kore.disambiguation.SmtSortParser.Sort(SmtSortParser.java:37)
	at org.kframework.parser.concrete2kore.disambiguation.TypeInferencer.readSort(TypeInferencer.java:827)
	at org.kframework.parser.concrete2kore.disambiguation.TypeInferencer.computeValue(TypeInferencer.java:816)
	at org.kframework.parser.concrete2kore.disambiguation.TypeInferencer.replayConstraints(TypeInferencer.java:210)
	at org.kframework.parser.concrete2kore.disambiguation.TypeInferencer.push(TypeInferencer.java:256)
	at org.kframework.parser.concrete2kore.disambiguation.TypeInferencer.error(TypeInferencer.java:760)
	at org.kframework.parser.concrete2kore.disambiguation.TypeInferenceVisitor.apply(TypeInferenceVisitor.java:115)
	at org.kframework.parser.concrete2kore.ParseInModule.parseStringTerm(ParseInModule.java:230)
	at org.kframework.parser.concrete2kore.ParseInModule.parseString(ParseInModule.java:150)
	at org.kframework.kompile.DefinitionParsing.performParse(DefinitionParsing.java:420)
	at org.kframework.kompile.DefinitionParsing.lambda$resolveNonConfigBubbles$15(DefinitionParsing.java:278)
	at java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:267)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ReduceOps$ReduceTask.doLeaf(ReduceOps.java:747)
	at java.util.stream.ReduceOps$ReduceTask.doLeaf(ReduceOps.java:721)
	at java.util.stream.AbstractTask.compute(AbstractTask.java:316)
	at java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:731)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.execLocalTasks(ForkJoinPool.java:1040)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1058)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
[Error] Internal: Uncaught exception thrown of type TokenMgrError (null)
make: *** [.build/defn/haskell/beacon-chain-kompiled/definition.kore] Error 1
```

--- a/kernel/src/main/java/org/kframework/parser/concrete2kore/disambiguation/TypeInferencer.java
+++ b/kernel/src/main/java/org/kframework/parser/concrete2kore/disambiguation/TypeInferencer.java
@@ -206,7 +206,7 @@ public class TypeInferencer implements AutoCloseable {
       case UNSATISFIABLE:
         println("(pop)");
         computeStatus();
-        if (constraint.isVar()) {
+        if (constraint.name != null) {
           Sort actualSort = computeValue(constraint.name);
           Sort expectedSort = eval(constraint.expectedSort, constraint.expectedParams);
           throw new LocalizedError("Unexpected sort " + actualSort + " for variable " + constraint.loc.value() + ". Expected: " + expectedSort, constraint.loc);

